node.go

```diff
@@ -4,13 +4,15 @@ import (
 	"context"
 	"fmt"
 
+	pubsub "github.com/libp2p/go-floodsub"
 	host "github.com/libp2p/go-libp2p-host"
 	peer "github.com/libp2p/go-libp2p-peer"
 	ma "github.com/multiformats/go-multiaddr"
 )
 
 type Node struct {
 	host.Host        // lib-p2p host
+	discovery        Discovery
```

Comment:
I don't see why `discovery` is an attribute both here and `ShardManager`. IMO it is better to put it only in one place. It seems `discovery` is only called by `ShardManager`. Because it is currently used to `Advertise`/`FindPeer` in `ShardID`, I think it is more suitable to put `discovery` in `ShardManager`. What do you think?

Reply:
I suppose if we do general topic discovery, it would be better to put `discovery` in `Node`?

Reply:
Agree. If this is the case, could we just put `discovery` in `Node`, and remove it from `ShardManager`?